### PR TITLE
ENG-453: Improvements of seo plugin (v5.3.0-dev)

### PR DESCRIPTION
**Pull request by eugeniosant**

### CODE REVIEW

File: plugins/entando-plugin-jpseo/src/main/java/org/entando/entando/plugins/jpseo/aps/system/services/mapping/SeoMappingDAO.java [hunk 1]
```diff
@@ -79,7 +79,7 @@ public void updateMapping(FriendlyCodeVO vo) {
 		try {
 			conn = this.getConnection();
 			conn.setAutoCommit(false);
-			this.deleteRecord(vo.getFriendlyCode(), vo.getPageCode(), vo.getContentId(), conn);
+			this.deleteRecord(vo.getFriendlyCode(), null, null, conn);
```
Comment (joewhite101) [inline plugins/entando-plugin-jpseo/src/main/java/org/entando/entando/plugins/jpseo/aps/system/services/mapping/SeoMappingDAO.java:82]: is this safe? Should the method signature be changed? What about records already in the DB? Do we need a migration script?

File: plugins/entando-plugin-jpseo/src/main/java/org/entando/entando/plugins/jpseo/aps/system/services/page/SeoPageMetadata.java [hunk 2]
```diff
@@ -156,6 +150,13 @@ public boolean hasEqualConfiguration(Object obj) {
             return check;
         }
         SeoPageMetadata other = (SeoPageMetadata) obj;
+        if (this.getFriendlyCode() == null) {
```
Comment (joewhite101) [inline plugins/entando-plugin-jpseo/src/main/java/org/entando/entando/plugins/jpseo/aps/system/services/page/SeoPageMetadata.java:153]: Please add a comment on what is happening from 153 to 178. Don't have to change it now but this whole section for this return is really really complex and should be refactored.

File: plugins/entando-plugin-jpseo/src/main/java/org/entando/entando/plugins/jpseo/aps/system/services/mapping/SeoMappingManager.java [hunk 3]
```diff
@@ -71,19 +72,21 @@ public void init() throws Exception {
...
+        if (!PageChangedEvent.EVENT_TYPE_SET_PAGE_OFFLINE.equals(event.getEventType()) && 
+                !PageChangedEvent.EVENT_TYPE_SET_PAGE_ONLINE.equals(event.getEventType())) {
+            return;
+        }
 		try {
 			if (event.getSource().equals(ApsEvent.LOCAL_EVENT)) {
-				if (null == this.getPageManager().getOnlinePage(page.getCode())) {
-					this.deleteMapping(page.getCode(), null);
-				} else {
+                this.deleteMapping(page.getCode(), null);
```
Comment (joewhite101) [inline plugins/entando-plugin-jpseo/src/main/java/org/entando/entando/plugins/jpseo/aps/system/services/mapping/SeoMappingManager.java:81]: Is this logical change intentional? Seems like an exception waiting to happen if the page isn't online? As long as it is covered by tests it is ok but want to check

**Comment by entando-service** [comment]
All is Well

File: plugins/entando-plugin-jpseo/src/main/java/org/entando/entando/plugins/jpseo/aps/system/services/mapping/SeoMappingDAO.java (hunk 1 above)
Comment (eugeniosant) [reply plugins/entando-plugin-jpseo/src/main/java/org/entando/entando/plugins/jpseo/aps/system/services/mapping/SeoMappingDAO.java]: I've done some other code refactoring and improvement of several methods. Can you check another time?

File: plugins/entando-plugin-jpseo/src/main/java/org/entando/entando/plugins/jpseo/aps/system/services/mapping/SeoMappingManager.java (hunk 3 above)
Comment (eugeniosant) [reply plugins/entando-plugin-jpseo/src/main/java/org/entando/entando/plugins/jpseo/aps/system/services/mapping/SeoMappingManager.java]: I've changed also this function

File: plugins/entando-plugin-jpseo/src/main/java/org/entando/entando/plugins/jpseo/aps/system/services/page/SeoPageMetadata.java (hunk 2 above)
Comment (eugeniosant) [reply plugins/entando-plugin-jpseo/src/main/java/org/entando/entando/plugins/jpseo/aps/system/services/page/SeoPageMetadata.java:153]: This method (extension of the core method) allows to understand if the current draft page is different then the related public page. This plugin adds other metadata to the page, so this core method needs to be extended

**Comment by entando-service** [comment]
All is Well

**Pull request merged**